src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -239,6 +278,12 @@ public Person toModelType() throws IllegalValueException {
 
         Person generatedPerson = modelRole.generatePerson(modelName,
                 modelPhone, modelEmail, modelAffiliations, modelAffiliationHistory);
+        if (generatedPerson instanceof Doctor) {
+            Doctor doctor = (Doctor) generatedPerson;
+            doctor.setSpecialisations(modelSpecialisations);
+            doctor = (Doctor) doctor.setShiftDays(modelShiftDays);
```

Comment:
Might be redundant to assign and typecasting it to `doctor` here?

Reply:
.setShiftDays(modelShiftDays); returns a Staff so I typecasted it again to preserve the Doctor type, but removing it seems to pass the tests so I'll heed ur advice, thanks!